src/Microsoft.Extensions.DependencyInjection.Specification.Tests/DependencyInjectionSpecificationTests.cs

```diff
@@ -158,6 +158,33 @@ public void MultipleServiceCanBeIEnumerableResolved()
         }
 
         [Fact]
+        public void RegistrationOrderIsPreservedWhenServicesAreIEnumerableResolved()
+        {
+            // Arrange
+            var collection = new ServiceCollection();
+            collection.AddTransient(typeof(IFakeMultipleService), typeof(FakeOneMultipleService));
+            collection.AddTransient(typeof(IFakeMultipleService), typeof(FakeTwoMultipleService));
+
+            var provider = CreateServiceProvider(collection);
+
+            collection.Reverse();
```

Comment:
Is this really reversing the collection? `ServiceCollection` only implements `IList<ServiceDescriptor>` and AFAIK that doesn't have a `Reverse` method.

That means this probably is the [`Enumerable.Reverse`](https://msdn.microsoft.com/en-us/library/bb358497(v=vs.100).aspx) LINQ method, which doesn't alter the original source, but returns a new `IEnumerable<ServiceDescriptor>`.